protocol/xmrmaker/event.go

```diff
@@ -237,9 +237,10 @@ func (s *swapState) handleEventContractReady() error {
 		log.Warnf("failed to claim funds from contract, attempting to safely exit: %s", err)
 
 		// TODO: retry claim, depending on error (#162)
-		if err = s.exit(); err != nil {
+		if err := s.exit(); err != nil { //nolint:govet
```

Comment:
I know it's uglier, but I would use another name or save the outer error in a variable. Intentionally using shadowing is too subtle. There is a good chance the next person to modify the code, not realizing its delicateness, will reintroduce the bug.

Reply:
updated